Python/optimizer_bytecodes.c

```diff
@@ -113,6 +115,23 @@ dummy_func(void) {
         sym_set_type(right, &PyLong_Type);
     }
 
+    op(_GUARD_TYPE_VERSION, (type_version/2, owner -- owner)) {
+        if (sym_matches_type_version(owner, type_version)) {
+            REPLACE_OP(this_instr, _NOP, 0, 0);
+        } else {
+            // add watcher so that whenever the type changes we invalide this
+            PyTypeObject *type = _PyType_LookupByVersion(type_version);
+            // if the type is null, it was not found in the cache (there was a conflict)
+            // with the key, in which case we can't trust the version
+            if (type) {
+                sym_set_type_version(owner, type_version);
+                PyType_Watch(TYPE_WATCHER_ID, (PyObject *)type);
```

Comment:
This needs to be checked for error.

Reply:
We don't check for the other watcher's erroring out either because they can never fail (we already reserved a slot for them). Mark's code for `PyDict_Watch` in this file doesn't check for error either.

Reply:
I added an assert here to verify this.